lib/src/services/databases/dataRefresh/hybrid.dart

```diff
@@ -0,0 +1,19 @@
+import "../hybrid.dart";
+import "implementation.dart";
+import "interface.dart";
+
+
+///
+// ignore: lines_longer_than_80_chars
+class HybridDataRefresh extends HybridDatabase<DataRefreshInterface> implements DataRefreshInterface{
+  @override
+  Future<void> signIn() async { }
+  @override
+  final CloudDataRefresh cloud = CloudDataRefresh();
+  @override
+  final LocalDataRefresh local = LocalDataRefresh();
+  @override
+  Future<void> getRefreshData() => cloud.getRefreshData();
+  @override
+  Future<void> setRefreshData(String data) => local.setRefreshData(data);
```

Comment:
```suggestion
  @override
  Future<void> signIn() async { }
  
  @override
  final CloudDataRefresh cloud = CloudDataRefresh();
  
  @override
  final LocalDataRefresh local = LocalDataRefresh();
  
  @override
  Future<void> getRefreshData() => cloud.getRefreshData();
  
  @override
  Future<void> setRefreshData(String data) => local.setRefreshData(data);
```

First of all, spacing. 

Secondly, notice that you separate the cloud and local databases by splitting them into `getRefreshData` and `setRefreshData`. That's good design within `HybridDataRefresh`, since you care about having a difference between the two. However, the function that uses the `HybridDataRefresh` (like `Database.init` or `Services.init`) won't care, and will just want the data to come from the right places automatically. 

That's the whole point of the `HybridX` interface. It's to compile the cloud and local databases into a single class that knows when to use which so that every other function can call `HybridX.doSomething()` without worrying. All this to say that you should have a function, maybe `Future<void> update(String data)`, that 1) downloads the refresh timestamp from the cloud, 2) compares it with the local timestamp,  3) updates the data if needed, and 4) save the current timestamp to the local database. 

You'll notice a few problems, namely: 
- if you download each refresh timestamp separately, you're essentially reading the same document many times for no reason
- how would `HybridDataRefresh` know how to update, say, `HybridCalendar` if it determines it needs to update the calendar? Each `HybridX` should be separate from the others (modular), so it wouldn't make sense to give it direct access to every other database. 

I think those two problems highlight the need for a better spot for this to go. Maybe you can give each `HybridX` a `.update` function, and in `Database.init`, you can download all the refresh timestamps at once and call any `.update` functions that are needed.

Reply:
wait .getRefreshData() and .setRefreshData() do two different things not something for just one local and the other on the cloud. The thing is the way I currently designed it, I don't need the ability to set the dates in the cloud from the app (at least for now). Also when Database.init() is called isn't the user not signed in yet?

Reply:
> wait .getRefreshData() and .setRefreshData() do two different things not something for just one local and the other on the cloud.

Of course, but the point is they're targeted at the local vs the cloud databases, and whoever uses `HybridDataRefresh` has to know that, which defeats the point of a `HybridX` class that can make the local vs cloud choice by itself. 

> The thing is the way I currently designed it, I don't need the ability to set the dates in the cloud from the app

So then you don't need both .get and .set in your interface, because the interface dictates what the local, cloud, and hybrid databases should share in common.

> Also when Database.init() is called isn't the user not signed in yet?

Right, there are two cases: 
1. the user is signed in, and `Database.init` can access Firestore
2. the user isn't signed in, `Database.init` cannot access Firestore and `Database.signIn` has to instead.

This whole feature only applies to case 1. If the user isn't signed in yet, you don't have to "refresh" anything, all the data will be downloaded in `signIn`. So you need an if statement in `Database.init` to check if the user is signed in (see the `Auth` class) and if so, see if the data needs to be updated.